### ISSUE TITLE
Breadcrumb not displaying correctly if URL has case mismatch

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#23803]
### Proposed Objective

Accessibility (WCAG)

### Proposed Priority

Priority 3 - Average

### dotCMS Version

21.06.12 LTS, demo (23.01)

### Problem Statement

If the request URI has a case mismatch, but still matches with a page, the page will render normally, but the breadcrumbs generated from NavTool do not generate properly.

### Steps to Reproduce

1- Create a code snippet widget with the following code:
```
<ul>
#set ($navCrumbs = $navtool.getNav("/"))
#foreach ($topNav in $navCrumbs)
#if ($topNav.active)
<li>$topNav.title
#if ($UtilMethods.isSet($topNav.children))
<ul>
#foreach ($secNav in $topNav)
#if ($secNav.active)
<li>$secNav.title</li>
#end
#end
</ul>
#end
</li>
#end
#end
</ul>
```
2- Add this code snippet to the `/store/outdoor/index` page.
3- Flush Cache.
4- On the front page hit the page (breadcrumb will display as expected).
5- Hit again the page but change the case of a letter e.g `/store/Outdoor/index`
6- See that the page will display but the breadcrumb is not displaying the `outdoor` item.

### Acceptance Criteria

Breadcrumb should resolve successfully since the page is rendering successfully.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

Customer ticket: https://dotcms.zendesk.com/agent/tickets/109671
